Description: virt-aa-helper: add a rule allowing all private channel access

The older libvirt in Trusty creates some channels under a namespace that
is not covered by virt-aa-helper. But since the scheme of these
channel files is known a rule can be added that still name-spaces per guest.
So always allow rw to things under that directory.

In latter Ubuntu releases the path changes and in even latter ones the delta
is dropped as the paths generated by libvirt now match those created by
virt-aa-helper.

Forwarded: no (solved by proper namespaceing and new virt-aa-helper rules)
Author: Christian Ehrhardt <christian.ehrhardt@canonical.com>
Original-Author: Serge Hallyn <serge.hallyn@ubuntu.com>
Origin: https://git.launchpad.net/~libvirt-maintainers/ubuntu/+source/libvirt/tree/debian/patches/ubuntu/virt-aa-helper-add-guest-agent-rule.patch?h=ubuntu/xenial
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1393842
Last-Update: 2017-08-28

--- a/src/security/virt-aa-helper.c
+++ b/src/security/virt-aa-helper.c
@@ -973,6 +973,10 @@ get_files(vahControl * ctl)
                                      ctl->def->parallels[i]->source.type) != 0)
                 goto cleanup;
 
+    virBufferAsprintf(&buf, "  # for qemu guest agent channel\n");
+    virBufferAsprintf(&buf, "  owner \"/var/lib/libvirt/qemu/channel/target/%s.**\" rw,\n",
+                      ctl->def->name);
+
     for (i = 0; i < ctl->def->nchannels; i++)
         if (ctl->def->channels[i] &&
             (ctl->def->channels[i]->source.type == VIR_DOMAIN_CHR_TYPE_PTY ||
